Code/Tools/AssetProcessor/native/utilities/BuilderManager.cpp

```diff
@@ -191,35 +191,24 @@ namespace AssetProcessor
         int portNumber = 0;
         ApplicationServerBus::BroadcastResult(portNumber, &ApplicationServerBus::Events::GetServerListeningPort);
 
-        AZStd::string params;
-#if !AZ_TRAIT_OS_PLATFORM_APPLE && !AZ_TRAIT_OS_USE_WINDOWS_FILE_PATHS
-        params = AZStd::string::format(
-            R"(-task=%s -id="%s" -project-name="%s" -project-cache-path="%s" -project-path="%s" -engine-path="%s" -port %d)",
-            task, builderGuid.c_str(), projectName.c_str(), projectCacheRoot.absolutePath().toUtf8().constData(),
-            projectPath.c_str(), enginePath.c_str(), portNumber);
-#else
-        params = AZStd::string::format(
-            R"(-task=%s -id="%s" -project-name="\"%s\"" -project-cache-path="\"%s\"" -project-path="\"%s\"" -engine-path="\"%s\"" -port %d)",
-            task, builderGuid.c_str(), projectName.c_str(), projectCacheRoot.absolutePath().toUtf8().constData(),
-            projectPath.c_str(), enginePath.c_str(), portNumber);
-#endif // !AZ_TRAIT_OS_PLATFORM_APPLE && !AZ_TRAIT_OS_USE_WINDOWS_FILE_PATHS
+        AZStd::vector<AZStd::string> params;
+        params.emplace_back(AZStd::string::format(R"(-task="%s")", task));
+        params.emplace_back(AZStd::string::format(R"(-id="%s")", builderGuid.c_str()));
+        params.emplace_back(AZStd::string::format(R"(-project-name="%s")", projectName.c_str()));
+        params.emplace_back(AZStd::string::format(R"(-project-cache-path="%s")", projectCacheRoot.absolutePath().toUtf8().constData()));
+        params.emplace_back(AZStd::string::format(R"(-project-path="%s")", projectPath.c_str()));
+        params.emplace_back(AZStd::string::format(R"(-engine-path="%s")", enginePath.c_str()));
+        params.emplace_back(AZStd::string::format("-port=%d", portNumber));
```

Comment:
This is a lot more readable when it comes to specifying arguments for a process.